### PR TITLE
add typed topic args

### PR DESCRIPTION
**Pull request by paulwe**
adds options for topic arg generation and batched registration/deregistration

### CODE REVIEW

File: protoc-gen-psrpc/options/options.proto [hunk 1]
```diff
@@ -20,9 +20,22 @@ message Options {
   // This method uses topics.
   bool topics = 3;
 
+  TopicParamOptions topic_params = 4;
+
   // Your service will supply an affinity function for handler selection.
-  bool affinity_func = 4;
+  bool affinity_func = 5;
```
Comment (biglittlebigben) [inline protoc-gen-psrpc/options/options.proto:26]: Will changing these ids break anything? Or is this message currently unused?

**Review by biglittlebigben — COMMENTED** [review]
Would it be possible to get some context about the usage of the Registerer?

**Comment by paulwe** [comment]
> Would it be possible to get some context about the usage of the Registerer?

the registerer is a bit like the client/server types, it's exported because the codegen uses it but it's not meant to be used directly.

`TopicParamOptions` allows changing the way topic arguments are generated. when a topic `group` is provided all rpc with the same group can be registered/deregistered together with `RegisterAllXYZTopics` and `DeregisterAllXYZTopics`. this avoids having to manually register and roll back partially successful registrations on errors.

for example in livekit all room service rpcs are grouped into `participant` and `room` topics (https://github.com/livekit/livekit/blob/signal-psrpc/pkg/service/rpc/gen/room.proto). when we register and deregister these in the app we only need one function call instead of one per rpc (https://github.com/livekit/livekit/blob/signal-psrpc/pkg/service/roommanager.go#L436, https://github.com/livekit/livekit/blob/signal-psrpc/pkg/service/roommanager.go#L458).

in the codegen `Registerer` is used to wrap the `RegisterXYZ` and `DeregisterXYZ` functions for each rpc (https://github.com/livekit/livekit/blob/signal-psrpc/pkg/service/rpc/gen/room.psrpc.go#L213-L218) so we can operate on them in a loop (https://github.com/livekit/psrpc/pull/29/files#diff-db42c1e8964070f0fb1482608e7dc99b2635428134237eff5cb3473f08eefe3bR22-R41). we're using runtime reflection here because the performance required from these functions is not demanding and the type safe solutions add considerable complexity with limited tangible benefits (https://gist.github.com/paulwe/230329c608fce8d4d51d4f71d0347e39, https://github.com/livekit/psrpc/pull/29/commits/02eceb8807101e8abd3e563f7244c8ec62e644c7#diff-283c70355a5a22d6576ae31b683adb77d72f6b9ce1de822ddb3f9f1117f54041).

**Review by biglittlebigben — COMMENTED** [review]
Seems reasonable to me. Somebody else with more experience with code gen code may want to have a look as well.

**Review by frostbyte73 — APPROVED** [review]

**Pull request merged**